src/main/java/com/google/devtools/build/remote/client/LogParserUtils.java

```diff
@@ -61,16 +62,35 @@ private FileInputStream openGrpcFileInputStream() throws ParamException, IOExcep
     return new FileInputStream(filename);
   }
 
+  // Returns an ExecutionResponse contained in the operation and null if none.
+  // If the operation contains an error, returns null and populates StringBuilder "error" with
+  // the error message.
+  public static <T extends Message> T getExecutionResponse(
+      Operation o, Class<T> t, StringBuilder error) throws IOException {
+    if (o.getResultCase() == ResultCase.ERROR && o.getError().getCode() != Code.OK.value()) {
```

Comment:
Note: even though leaving this check is okay, this should never happen. The API specifically instructs to not use the ERROR case, and return statuses in the ExecuteResponse instead. This is not just a V2 change, it was a very early amendment to V1. So I'd also be okay with a Precondition failure in this case.